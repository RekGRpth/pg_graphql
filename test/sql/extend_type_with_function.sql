begin;

    create table public.account(
        id serial primary key,
        first_name varchar(255) not null,
        last_name varchar(255) not null
    );

    -- Extend with function
    create function public._full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select format('%s %s', rec.first_name, rec.last_name)
    $$;

    insert into public.account(first_name, last_name)
    values
        ('Foo', 'Fooington'),
        ('Bar', 'Barsworth');


    savepoint a;

    select jsonb_pretty(
        graphql.resolve($$
    {
      accountCollection {
        edges {
          node {
            id
            firstName
            lastName
            fullName
          }
        }
      }
    }
        $$)
    );

    rollback to savepoint a;

    select graphql.resolve($$
    {
      account(id: "WyJhY2NvdW50IiwgMV0=") {
        id
        fullName
      }
    }
    $$);

    rollback to savepoint a;

rollback;
