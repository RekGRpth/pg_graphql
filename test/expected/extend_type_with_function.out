begin;
    create table public.account(
        id serial primary key,
        first_name varchar(255) not null,
        last_name varchar(255) not null
    );
    -- Extend with function
    create function public._full_name(rec public.account)
        returns text
        immutable
        strict
        language sql
    as $$
        select format('%s %s', rec.first_name, rec.last_name)
    $$;
    insert into public.account(first_name, last_name)
    values
        ('Foo', 'Fooington'),
        ('Bar', 'Barsworth');
    savepoint a;
    select jsonb_pretty(
        graphql.resolve($$
    {
      accountCollection {
        edges {
          node {
            id
            firstName
            lastName
            fullName
          }
        }
      }
    }
        $$)
    );
                     jsonb_pretty                     
------------------------------------------------------
 {                                                   +
     "data": {                                       +
         "accountCollection": {                      +
             "edges": [                              +
                 {                                   +
                     "node": {                       +
                         "id": 1,                    +
                         "fullName": "Foo Fooington",+
                         "lastName": "Fooington",    +
                         "firstName": "Foo"          +
                     }                               +
                 },                                  +
                 {                                   +
                     "node": {                       +
                         "id": 2,                    +
                         "fullName": "Bar Barsworth",+
                         "lastName": "Barsworth",    +
                         "firstName": "Bar"          +
                     }                               +
                 }                                   +
             ]                                       +
         }                                           +
     },                                              +
     "errors": [                                     +
     ]                                               +
 }
(1 row)

    rollback to savepoint a;
    select graphql.resolve($$
    {
      account(id: "WyJhY2NvdW50IiwgMV0=") {
        id
        fullName
      }
    }
    $$);
                                   resolve                                   
-----------------------------------------------------------------------------
 {"data": {"account": {"id": 1, "fullName": "Foo Fooington"}}, "errors": []}
(1 row)

    rollback to savepoint a;
rollback;
